### PR TITLE
roachtest: --use-spot now properly defaults to "no SpotVMs"

### LINKED ISSUE #77376: linter: Add PR check to encourage including issue or epic refs in the PR/commit messages
**Describe the solution you'd like**

This new linter should be a program run by a GitHub action on every PR for these events: create, edit, reopen, synchronize.

What it will do:

For all PRs containing commits with [Release note texts](https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/186548364/Release+notes), other than bug fixes, Engineers can help establish the link to Jira epics in various ways:

- If the PR relates to only one Jira epic:
    - Put the GH issue or Jira epic reference either in the PR description or in each relevant commit.
    - If the reference is in the PR description, put “Epic: None” in each commit that doesn’t relate to the epic. This is particularly important if the commit contains a release note.
- If the PR relates to multiple Jira epics:
    - Put the GH issue or Jira epic reference in each relevant commit.
    - If a single commit relates to multiple Jira epics, put all relevant GH issue or Jira epic references in that commit.
    - If all commits relate to all the Jira epics, put the GH issue or Jira epic references in the PR description or in the commit messages.
    - Put “Epic: None” in each commit with a release note that doesn’t relate to any epic.
- If the PR relates to no Jira epics:
    - Preferably, put “Epic: None” in the PR description or in each commit containing a release note.

Given Engineering’s practice to [link GitHub issues to epics](https://cockroachlabs.atlassian.net/wiki/spaces/JIRA/pages/1925283849/Epic+Linking), referencing such an issue—as opposed to the epic it links to—is preferred in cases where both a Jira epic and GitHub issue exist. This makes information in the issue easily accessible to human readers while offering an epic link that is available via the issue to automation scripts.

The “Epic: None” option accounts for the fact that a significant amount of product changes occur spontaneously (serendipitously) as a by-product of investigations on main roadmap items. When this happens, Engineers take the initiative to make user-facing changes without prior processing by a PM, and thus without prior existence of a GH or Jira issue. We do not wish to reduce the ability of Engineers to innovate and improve our product in this way. 

This solution will be encouraged, but not required, by a new optional blathers check on PRs. It will fail and give feedback on how to adjust the PR for the following cases:

- When no GH issue reference, Jira epic reference, or “Epic: None” exists in the PR description or commit messages.
- When multiple Jira epic references are in the PR description but not in individual commit messages.
- When multiple Jira epic references are in a commit message (give a warning that it’s not a common case and to check that they intend it).
- When backport PRs backport multiple source PRs and one or more of the source PRs has one of the above cases.

**Describe alternatives you've considered**

This linter can't run in TeamCity because it needs to run on PR edit and reopen events. We considered using Blathers but decided to keep this in the cockroach repo.

Jira issue: CRDB-13558

Epic DEVINF-261

### PR DESCRIPTION
**Pull request by srosenberg**
In [1], the default of `UseSpotVM` was reinterpreted to mean
the framework decides whether or not to create a spot VM.
This may impede local (performance) testing when a potential
preemption is unwelcome; i.e., may result in loss of work.

This change reverts the original default of "no SpotVMs".
It now interprets `--use-spot auto` to mean that the framework
decides whether or not to create a spot VM, whereas `--use-spot never`
(default) unconditionally disables the use of spot VMs.
Furthermore, to allow unconditional creation of spot VMs, pass
`--use-spot always`. Finally, when running in CI,
we pass `--use-spot auto`, at this time, to allow the framework
to conditionally create spot VMs, i.e., based on current heuristics.

[1] https://github.com/cockroachdb/cockroach/pull/116778

Epic: none
Fixes: #77376
Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/roachtestflags/flags.go [hunk 1]
```diff
@@ -359,8 +359,18 @@ var (
...
-		Name:  "use-spot",
-		Usage: `Use SpotVM to run tests, If the provider does not support spotVM, it will be ignored`,
+		Name: "use-spot",
+		Usage: `
+			May use SpotVM to run tests; note, this is merely a _hint_. The framework decides if a SpotVM should be used. 
+			You can override by passing --use-spot-always.`,
+	})
+
+	AlwaysUseSpotVM bool
+	_               = registerRunFlag(&UseSpotVM, FlagInfo{
```
Comment (DarrylWong) [inline pkg/cmd/roachtest/roachtestflags/flags.go:369]: `&AlwaysUseSpotVM` instead 😄

**Review by DarrylWong — APPROVED** [review]
LGTM after you fix the one small issue!

File: pkg/cmd/roachtest/roachtestflags/flags.go (hunk 1 above)
Comment (srosenberg) [reply pkg/cmd/roachtest/roachtestflags/flags.go:369]: Oops, that would have been frustrating to someone flooring the gas pedal :)

**Comment by srosenberg** [comment]
Tested the options `--use-spot=true`, `--use-spot=false` and `--use-spot-always=true` on my gceworker; no issues found.

Triggered [GCE Run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15287726?buildTab=log&logView=linear&focusLine=34) in TC, with `SELECT_PROBABILITY=0.6` as a sanity check.

**Review by nameisbhaskar — COMMENTED** [review]
Thanks for the change! Just a suggestion, can we not make it in a single flag with enumeration? `always`, `default`, `never`?

**Comment by srosenberg** [comment]
> Thanks for the change! Just a suggestion, can we not make it in a single flag with enumeration? `always`, `default`, `never`?

We could except we want the `default` to be `never`. Ideally, we also want to keep the original flag for backward compatibility. Maybe `always`, `never` (default), and `maybe`? :)

**Comment by nameisbhaskar** [comment]
> > Thanks for the change! Just a suggestion, can we not make it in a single flag with enumeration? `always`, `default`, `never`?
> 
> We could except we want the `default` to be `never`. Ideally, we also want to keep the original flag for backward compatibility. Maybe `always`, `never` (default), and `maybe`? :)

got it. actually, it should be just a boolean. We are having this flag just because of the 50% condition that we have today.
Basically, If a specific test sets UseSpot true, this new flag can override that going forward. So, it may be ok to keep to flags and remove "use-spot" later?

**Review by herkolategan — APPROVED** [review]

**Comment by srosenberg** [comment]
> So, it may be ok to keep to flags and remove "use-spot" later?

We won't be able to remove "use-spot" because we need to allow for disabling spot usage, by default and on request.

**Review by nameisbhaskar — APPROVED** [review]

**Comment by srosenberg** [comment]
Removed `--use-spot-always` and augmented `--use-spot [never|always|auto]` as discussed. PTAL!

**Review by DarrylWong — APPROVED** [review]
Not going with surprise? 🫨 

LGTM!

**Comment by srosenberg** [comment]
TFTR!

bors r=DarrylWong,herkolategan,nameisbhaskar

**Pull request merged**